apps/ledger-live-mobile/src/components/Stake/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { useMemo, useLayoutEffect, useCallback } from "react";
+import { useNavigation } from "@react-navigation/native";
+import { Account, AccountLike } from "@ledgerhq/types-live";
+import {
+  listCurrencies,
+  filterCurrencies,
+} from "@ledgerhq/live-common/currencies/helpers";
+import { useFeature } from "@ledgerhq/live-common/featureFlags/index";
+import { NavigatorName, ScreenName } from "../../const";
+import perFamilyAccountActions from "../../generated/accountActions";
+
+const StakeFlow = ({ currencies }) => {
+  const { params: { list } = { list: undefined } } =
+    useFeature("stakePrograms") || {};
+  const navigation = useNavigation();
+  const cryptoCurrencies = useMemo(() => {
+    return filterCurrencies(listCurrencies(true), {
+      currencies: currencies || list || [],
+    });
+  }, [currencies, list]);
+
+  const onSuccess = useCallback(
+    (account: AccountLike, parentAccount?: Account) => {
+      const decorators = perFamilyAccountActions[account?.currency?.family];
+      const familySpecificMainActions =
+        (decorators &&
+          decorators.getMainActions &&
+          decorators.getMainActions({
+            account,
+            parentAccount,
+            colors: {},
+          })) ||
+        [];
+      const stakeFlow = familySpecificMainActions?.find(
+        action => action.id === "stake",
+      )?.navigationParams;
+      const [name, options] = stakeFlow;
```

Comment:
`stakeFlow` could be undefined here - what should happen if we can't find it? early return maybe?